### DISCUSSION TITLE
ImmortalWrt Image Builder 使用说明

### DISCUSSION THREAD

**Discussion by 1715173329** [1715173329/blog#17] [Comments]
https://blog.imouto.in/post/openwrt/2023/immortalwrt-ib-usage/

**Comment by 0xffffharry** [comment]
是否可以根据dts调整布局

**Comment by EkkoG** [comment]
> 多谢博主的教程，很有启发，本来打算在 GitHub actions 上使用 ib 的 docker 镜像构建，奈何权限问题恶心。
> 
> 给友友们来个示例吧：https://github.com/douglarek/build-immortalwrt 。

可以选择用 ImageBuilder 的 docker 镜像来构建，不用处理依赖 https://github.com/EkkoG/OpenWrt/blob/master/.github/workflows/main.yml

**Comment by GoogleBeEvil** [comment]
有一个问题： uci-defaults  这里 想设置路由默认登陆ip为192.168.111.1，是需要在files文件夹下创建对应的层级目录
` files/etc/uci-defaults/99-custom`然后把一下内容写入99-custom么
```
lan_ip_address="192.168.111.1"
if [ -n "$lan_ip_address" ]; then
  uci set network.lan.ipaddr="$lan_ip_address"
  uci commit network
fi
echo "All done!"
```

**Comment by lambdaplus** [comment]
> 有一个问题： uci-defaults 这里 想设置路由默认登陆ip为192.168.111.1，是需要在files文件夹下创建对应的层级目录 ` files/etc/uci-defaults/99-custom`然后把一下内容写入99-custom么
> 
> ```
> lan_ip_address="192.168.111.1"
> if [ -n "$lan_ip_address" ]; then
>   uci set network.lan.ipaddr="$lan_ip_address"
>   uci commit network
> fi
> echo "All done!"
> ```

我是这样写的

```
# set lan ip
uci batch << EOF

set network.lan.ipaddr='192.168.111.1'

commit
EOF
```
